### DISCUSSION TITLE
Is there functionality to turn off notifications in app for Android?

### DISCUSSION THREAD

**Discussion by amiceli-torstar** [invertase/react-native-firebase#5498] [Q&A]
See title. Was wondering if there was functionality available where we could implement a toggle to turn on/off notifications in Android

**Comment by piljac1** [comment]
You can make it work with topic subcriptions. You can use `messaging().subscribeToTopic(topicKey)` to receive notifications sent to this topic and `messaging().unsubscribeFromTopic(topicKey)` to stop receiving notifications related to that topic.

**Reply by mikehardy** [reply]
Yep - @piljac1 is right, alternatively, if you `messaging.deleteToken()` (for instance, if a user logs out) then FCM delivery will fail (something you can detect in cloud functions, and remove the key from your backend) and user will no longer receive FCM if you're doing specific token addressing

**Comment by amiceli-torstar** [comment]
Appreciate the answers, I don't think I was clear in my question, I specifically meant turning on/off all notifications (equivalent to turning on and off in the Android settings).

**Reply by mikehardy** [reply]
At the app level, on Android, I don't think so - once you create a notification channel it's in the user's control so you can't alter the way notifications display. And if you send a notification to a device with a notification block, the underlying SDKs will display it.

So it would have to be a toggle where at the same time you stopped posting any local notification from the app yourself, you also did something on the back-end to stop sending FCM with notification payload to the client.

I could be wrong, but that's my understanding of the APIs available and how they behave

**Reply by amiceli-torstar** [reply]
That's good to know. Alternatively (or somewhat related I guess), could we check via the library whether the user currently has notifications enabled (not necessarily set the settings itself)?

**Reply by mikehardy** [reply]
I'm not entirely sure - this seems like something more oriented towards either react-native-permissions and/or a real notifications package that would let you inspect notification channel settings like these APIs here https://notifee.app/react-native/docs/android/channels#listening-to-channel-events

**Reply by mikehardy** [reply]
(note that I had to search that, I picked Notifee but I'm always aware it requires a license, I'm just not familiar with react-native-push-notifications - the other main package and it may or may not have similar APIs but it should be an easy check for you if that's interesting)
